api/team.go

```diff
@@ -192,77 +125,50 @@ func isTeamCreationAllowed(c *Context, email string) bool {
 }
 
 func GetAllTeamListings(c *Context, w http.ResponseWriter, r *http.Request) {
-	if result := <-app.Srv.Store.Team().GetAllTeamListing(); result.Err != nil {
-		c.Err = result.Err
+	var teams []*model.Team
+	var err *model.AppError
+
+	if teams, err = app.GetAllOpenTeams(); err != nil {
+		c.Err = err
 		return
-	} else {
-		teams := result.Data.([]*model.Team)
-		m := make(map[string]*model.Team)
-		for _, v := range teams {
-			m[v.Id] = v
-			if !HasPermissionToContext(c, model.PERMISSION_MANAGE_SYSTEM) {
-				m[v.Id].Sanitize()
-			}
-			c.Err = nil
-		}
+	}
 
-		w.Write([]byte(model.TeamMapToJson(m)))
+	m := make(map[string]*model.Team)
+	for _, v := range teams {
+		m[v.Id] = v
+		if !HasPermissionToContext(c, model.PERMISSION_MANAGE_SYSTEM) {
+			m[v.Id].Sanitize()
+		}
+		c.Err = nil
 	}
+
+	w.Write([]byte(model.TeamMapToJson(m)))
 }
 
 // Gets all teams which the current user can has access to. If the user is a System Admin, this will be all teams
 // on the server. Otherwise, it will only be the teams of which the user is a member.
 func getAll(c *Context, w http.ResponseWriter, r *http.Request) {
-	var tchan store.StoreChannel
+	var teams []*model.Team
+	var err *model.AppError
+
 	if HasPermissionToContext(c, model.PERMISSION_MANAGE_SYSTEM) {
-		tchan = app.Srv.Store.Team().GetAll()
+		teams, err = app.GetAllTeams()
```

Comment:
Nothing to do here just wanted to say that the validation of permissions here makes totally sense